pygmt/clib/conversion.py

```diff
@@ -263,14 +267,15 @@ def sequence_to_ctypes_array(
     return (ctype * size)(*sequence)
 
 
-def strings_to_ctypes_array(strings: Sequence[str]) -> ctp.Array:
+def strings_to_ctypes_array(strings: Sequence[str] | pa.StringArray) -> ctp.Array:
```

Comment:
Need to figure out how to fix the `mypy` type errors for an optional import :thinking:

```
pygmt/clib/conversion.py:298: error: Item "Sequence[str]" of "Sequence[str] | Any" has no attribute "to_pylist"  [union-attr]
pygmt/clib/session.py:1004: error: Item "Sequence[str]" of "Sequence[str] | Any" has no attribute "dtype"  [union-attr]
```